File: lecture/inflearn-spring-boot/autoconfig/src/test/java/hello/repository/MemberRepositoryTest.java
package hello.repository;

import hello.member.Member;
import static org.junit.jupiter.api.Assertions.*;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.transaction.annotation.Transactional;

@SpringBootTest
class MemberRepositoryTest {

  @Autowired
  MemberRepository memberRepository;

  @Transactional
  @Test
  void memberTest() {
    Member member = new Member("m1", "member1");
    memberRepository.initTable();
    memberRepository.save(member);
    Member findMember = memberRepository.find(member.getMemberId());

    assertEquals(member.getMemberId(), findMember.getMemberId());
  }

}

File: toy/kotlin-springboot-toy/src/test/kotlin/com/example/ex/repository/MemberRepositoryTest.kt
package com.example.ex.repository

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase
import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest

@AutoConfigureTestDatabase(
    replace = AutoConfigureTestDatabase.Replace.NONE
)
@DataJpaTest
internal class MemberRepositoryTest @Autowired constructor(private val memberRepository: MemberRepository) {

    @Test
    fun test() {
        val member = memberRepository.findById(1).get()
        println("member=${member.id}")
        assertEquals(1, memberRepository.count())
    }
}